pkg/server/commands/list_objects.go

```diff
@@ -104,13 +113,16 @@ func NewListObjectsQuery(ds storage.RelationshipTupleReader, opts ...ListObjects
 		listObjectsMaxResults:   defaultListObjectsMaxResults,
 		resolveNodeLimit:        defaultResolveNodeLimit,
 		resolveNodeBreadthLimit: defaultResolveNodeBreadthLimit,
+		maxConcurrentReads:      defaultMaxConcurrentReadsForListObjects,
 		checkOptions:            []graph.LocalCheckerOption{},
 	}
 
 	for _, opt := range opts {
 		opt(query)
 	}
 
+	query.datastore = storagewrappers.NewBoundedConcurrencyTupleReader(query.datastore, query.maxConcurrentReads)
```

Comment:
I want to point out that this datastore will be passed to `NewLocalChecker`, which is also doing `checker.ds = storagewrappers.NewBoundedConcurrencyTupleReader(checker.ds, checker.maxConcurrentReads)`

I guess it's fine though.

Reply:
It is expected that the caller constructs the derivative datastore. That is why we accept it as part of the constructor. The datastore is supposed to be a caller provided dependency. Otherwise we're wasting cycles and heap allocations for no reason.

Since these commands in the commands package are always nested with the top-level RPC, they will always have the request context available to them. There is no reason why we need to construct a duplicate derivative datastore.

```go
package server

func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequest) (*openfgav1.ListObjectsResponse, error) {
    boundedDatastore := storagewrappers.NewBoundedConcurrencyTupleReader(...)

    contextualDatastore := storagewrappers.NewContextualTupleReader(boundedDatastore, req.GetContextualTuples())
    
    checkResolver := NewLocalChekcer(contextualDatastore, ...)
    listObjectsQuery := NewListObjectsQuery(contextualDatastore, ...)

    return listObjectsQuery.Execute(...)
}
```

Reply:
Right now the checkResolver is not defined in the caller layer.  @miparnisari / @jon-whit - want to confirm whether we want to move the check resolver to server ListObjects / StreamedListObjects  method instead of being constructed in list object

Reply:
Maybe these types of refactoring should be done all together with all of us agreeing on the direction.

Reply:
Yeah, what I suggested needs some further discussion before we go about doing it at this time. I'll schedule some time for us to talk about "future state". For now, this follows status quo.